pkg/hostagent/dns/dns_test.go

```diff
@@ -0,0 +1,115 @@
+package dns
+
+import (
+	"io"
+	"log"
+	"net"
+	"regexp"
+	"testing"
+
+	"github.com/foxcpp/go-mockdns"
+	"github.com/miekg/dns"
+	"github.com/stretchr/testify/assert"
+)
+
+var (
+	dnsResult *dns.Msg
+)
+
+func TestTXTRecords(t *testing.T) {
+	testDomains := make([]string, 3)
+	testDomains[0] = "onerecord.com"
+	testDomains[1] = "multistringrecord.com"
+	testDomains[2] = "multiplerecords.com"
+
+	expectedResults := make([]string, 3)
+	expectedResults[0] = `onerecord.com.\s*5\s*IN\s*TXT\s*"My txt record"`
+	expectedResults[1] = `multistringrecord.com.\s*5\s*IN\s*TXT\s*"123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345678901234567890123456789012345" "67890123456789012345678901234567890"`
+	expectedResults[2] = `multiplerecords.com.\s*5\s*IN\s*TXT\s*"record 1"\nmultiplerecords.com.\s*5\s*IN\s*TXT\s*"record 2"`
```

Comment:
It doesn't matter much, but all `\s*` patterns should be `\s+` so they don't match the empty string. I believe at least a single space is required in each position.

Reply:
Good call.